Upgrade RSA Key from 1024 bit to 4096 to eliminate warnings

FrankChen021 opened the pull request:
### Description

Current Integration test uses 1024-bit RSA key to generate certificates for TLS. There're warnings when generating such RSA keys or running docker images. The warnings show as follows:

```
<druid> uses a 1024-bit RSA key which is considered a security risk. This key size will be disabled in a future update.
```

These warnings are noisy information when handling IT test case build failures such as #11663 . So I upgrade the length of RSA key from 1024 to 4096 to eliminate all of these warnings.

There're also wanings about use of keystore format as follows

```bash
The JKS keystore uses a proprietary format. It is recommended to migrate to PKCS12 which is an industry standard format using "keytool -importkeystore -srckeystore client.jks -destkeystore client.jks -deststoretype pkcs12".
```

I also change the format of keystore from JKS to PKCS12 to eliminate these warnings.


This PR has:
- [X] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

capistrant commented:
Seems like a smart change. curious as to why `druid.server.https.keyStoreType` didn't need to be updated in the IT configs when you switched to `pkcs12` as the type? From what I see in the IT env configs, it is `jks`. Perhaps I am misunderstanding what this config is doing

FrankChen021 commented:
@capistrant Thanks for pointing out that. I checked the tests and found it a little strange that even the keyStoreType keeps unchanged as `jks`, all the security tests passed. I don't know why.

But to keep it consistent, I changed the keyStoreType to PKCS12 and removed the shell command that exports the JKS keystore.

a2l007 reviewed (approved):
👍
